### PR TITLE
Add information about templates

### PR DESCRIPTION
**Pull request by ldevulder**
Add links to some docs as well as an example.

This should fix https://github.com/rancher-sandbox/cOS-toolkit/issues/1162.

### CODE REVIEW

File: content/en/docs/Reference/cloud_init.md [hunk 1]
```diff
@@ -102,6 +102,23 @@ EOF
...
 _Note_: Each cloud-init option can be either run in *dot notation* ( e.g. `stages.network[0].authorized_keys.user=github:user` ) in the boot args or either can supply a cloud-init URL at boot with the `cos.setup=$URL` parameter.
 
+### Using templates
+
+With Cloud Init support, templates can be used to allow dynamic configuration. For example in mass-install scenario it could be needed (and easier) to specify hostnames for multiple machines from a single cloud-init config file.
+
+```yaml
+#cloud-config
+hostname: node-{{ trunc 4 .MachineID }}
+install:
```
Comment (mudler) [inline content/en/docs/Reference/cloud_init.md:112]: install it's not really part of the standard cloud-init syntax

File: content/en/docs/Reference/cloud_init.md [hunk 2]
```diff
@@ -102,6 +102,23 @@ EOF
...
+  device: /dev/sda
```
Comment (mudler) [inline content/en/docs/Reference/cloud_init.md:113]: ditto here

File: content/en/docs/Reference/cloud_init.md (hunk 1 above)
Comment (ldevulder) [reply content/en/docs/Reference/cloud_init.md:112]: Ok, will remove this part.

**Review by mudler — APPROVED** [review]

**Comment by mudler** [comment]
Looks good here!

**Pull request merged**